src/make_trade.ts

```diff
@@ -0,0 +1,82 @@
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { HardhatEthersHelpers } from "@nomiclabs/hardhat-ethers/types";
+import ERC20 from "@openzeppelin/contracts/build/contracts/ERC20.json";
+import { TokenInfo, TokenList } from "@uniswap/token-lists";
+import { BigNumber, Contract } from "ethers";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+import fetch from "node-fetch";
+
+async function makeTrade(
+  tokenListUrl: string,
+  { ethers }: HardhatRuntimeEnvironment
+): Promise<void> {
+  const [trader] = await ethers.getSigners();
+
+  const all_tokens = (await fetchTokenList(tokenListUrl)).tokens;
```

Comment:
Not an issue, but aesthetically I find the following slightly nicer.
```ts
const { tokens: all_tokens } = await fetchTokenList(...)
``` 

Its 100% fine as is though.

Reply:
I'm changing fetchTokenList to return a TokenInfo[] directly as it turns out we need to filter tokens with other chainIds.